freqtrade/persistence/models.py

```diff
@@ -259,9 +263,79 @@ class LocalTrade():
     strategy: str = ''
     timeframe: Optional[int] = None
 
+    # Margin trading properties
+    borrowed_currency: str = None
+    collateral_currency: str = None
+    interest_rate: float = 0.0
+    liquidation_price: float = None
+    is_short: bool = False
+    borrowed: float = 0.0
+    leverage: float = None
+
+    # @property
+    # def base_currency(self) -> str:
+    #     if not self.pair:
+    #         raise OperationalException('LocalTrade.pair must be assigned')
+    #     return self.pair.split("/")[1]
+
+    # TODO: @samgermain: Amount should be persisted "as is".
+    # I've partially reverted this (this killed most of your tests)
+    # but leave this here as i'm not sure where you intended to use this.
+    # @property
+    # def amount(self) -> float:
+    #     if self._leverage is not None:
+    #         return self._amount * self.leverage
+    #     else:
+    #         return self._amount
+
+    # @amount.setter
+    # def amount(self, value):
+    #     self._amount = value
+
+    # @property
+    # def borrowed(self) -> float:
+    #     if self._leverage is not None:
+    #         if self.is_short:
+    #             # If shorting the full amount must be borrowed
+    #             return self._amount * self._leverage
+    #         else:
+    #             # If not shorting, then the trader already owns a bit
+    #             return self._amount * (self._leverage-1)
+    #     else:
+    #         return self._borrowed
+
+    # @borrowed.setter
+    # def borrowed(self, value):
+    #     self._borrowed = value
+    #     self._leverage = None
+
+    # @property
+    # def leverage(self) -> float:
+    #     return self._leverage
+
+    # @leverage.setter
+    # def leverage(self, value):
+    #     self._leverage = value
+    #     self._borrowed = None
```

Comment:
There's 0 reason to use a property here - that'll only complicate things with the database (the database column will be named differently ...).
Also - leverage will not change once the trade is created - you either use leverage - or you don't.